### PR TITLE
TP-1085, TP-1093: Making the plugin agnostic to framework dependencies

### PR DESCRIPTION
**Pull request by maheshsubramanian**

### CODE REVIEW

File: src/main/java/uk/gov/justice/plugin/validator/AnnotationValidator.java [hunk 1]
```diff
@@ -8,4 +8,6 @@
 public interface AnnotationValidator {
 
     ValidationResult validate(final Class<?> classWithAnnotation, final Map<String, String> validationProperties) throws ValidationException;
+
+    String worksOnAnnotatedClass();
```
Comment (matt-rich) [inline src/main/java/uk/gov/justice/plugin/validator/AnnotationValidator.java]: not sure if there is a better name than this.
Something like appliesToAnnotation maybe? not sure...

Comment (maheshsubramanian) [reply src/main/java/uk/gov/justice/plugin/validator/AnnotationValidator.java]: @matt-rich will change it to what you suggest..

File: src/main/java/uk/gov/justice/plugin/validator/AnnotationValidator.java [hunk 2]
```diff
@@ -8,4 +8,6 @@
 public interface AnnotationValidator {
```
Comment (matt-rich) [inline src/main/java/uk/gov/justice/plugin/validator/AnnotationValidator.java]: Just wondering if we can do this a smarter way and avoid having to implement the method.

Does this look any cleaner?
`public interface AnnotationValidator<T> {

   // We can remove this one now...
    String worksOnAnnotatedClass();

    default String getApplicableAnnotationName() {
        return this.getClass().getTypeParameters().getClass().getName();
    }
}`

File: src/test/java/uk/gov/justice/plugin/annotation/AnnotationCheckMojoTest.java [hunk 3]
```diff
@@ -81,19 +83,29 @@ public void testMultipleReports() throws Exception {
...
+
+        mojo.execute();
+
+        // assert report generation and content
+        verifyReport("dummy message");
     }
 
-    private void verifyReport() throws IOException {
-        final List<String> reportLines = readLines(new File(getQualifiedPathForReport(Event.class)));
+    private void verifyReport(String validationFailureMessageText) throws IOException {
```
Comment (matt-rich) [inline src/test/java/uk/gov/justice/plugin/annotation/AnnotationCheckMojoTest.java]: final

File: src/test/java/uk/gov/justice/plugin/validator/AnnotationValidatorFactoryTest.java [hunk 4]
```diff
@@ -1,26 +1,35 @@
...
 
+    private ClassLoader classLoader;
+
+    @Before
+    public void setup() {
+        classLoader = getClass().getClassLoader();
+    }
+
     @Test
     public void getValidatorForEventAnnotationType() {
```
Comment (matt-rich) [inline src/test/java/uk/gov/justice/plugin/validator/AnnotationValidatorFactoryTest.java]: rename as we are not using EventAnnotation anymore?

File: src/test/resources/annotation-plugin/pom-generate-report-with-additional-validator-properties.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                    </validatorProperties>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>uk.gov.justice.framework-api</groupId>
+            <artifactId>framework-api-domain</artifactId>
+            <version>2.0.3</version>
```
Comment (matt-rich) [inline src/test/resources/annotation-plugin/pom-generate-report-with-additional-validator-properties.xml]: should this use a property?

File: src/test/resources/annotation-plugin/pom-generate-report.xml [hunk 6]
```diff
@@ -13,14 +13,13 @@
             <plugin>
                 <groupId>uk.gov.justice.maven</groupId>
                 <artifactId>annotation-validator-maven-plugin</artifactId>
-                <version>1.6.6-SNAPSHOT</version>
+                <version>1.0.0-SNAPSHOT</version>
```
Comment (matt-rich) [inline src/test/resources/annotation-plugin/pom-generate-report.xml:16]: it's a bit difficult to see from just github. But should these 1.0.0-SNAPSHOT all be ${project.version} ? or a property or left as they are?

Just wondering if they will auto version with a maven release.

File: src/test/java/uk/gov/justice/plugin/annotation/AnnotationCheckMojoTest.java (hunk 3 above)
Comment (maheshsubramanian) [reply src/test/java/uk/gov/justice/plugin/annotation/AnnotationCheckMojoTest.java]: done

**Pull request merged**